compiler/src/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp

```diff
@@ -267,7 +312,7 @@ void LinalgFusePass::runOnOperation() {
   FailureOr<linalg::LinalgTilingAndFusionOptions> defaultTilingOptions =
       getTileAndFuseOptionsFromConfig(funcOp, tilingLevel);
   if (failed(defaultTilingOptions)) {
-    return signalPassFailure();
+    return;
```

Comment:
It seems to create problems downstream. I am guessing that signalPassFailure() kills the entire pipeline, not just this pass.

Reply:
`getTileAndFuseOptionsFromConfig` also fails if there's no LoweringConfig defined in the vicinity (I don't know what the vicnity is, but when I remove the `lowering_config` from the op, it fails; I want it to not fail then). (It looks like you have assumed `getTileAndFuseOptionsFromConfig` fails only when tiling_level is `-1`.)

Reply:
We use the lowering_config to drive all sandbox fine-grained transforms. After exploring all the code and logic, I think I understand what's happening here. The two generic ops created by SplitReduction, and might both of them carry the configuration. While getRootOp assumes that only one config can be existed, it return a failure().

IMO, silently skipping the pass is confusing. To address the issue, we should either

1) Fix the getRootOp to check if the configs are identical. It might raise other issues because you always run tiling on reductions right after SplitReduction.
2) Signal something when skipping the pass.

Though the fix hides the actual issue, I'm fine with the current change, since I can see the CodegenDriver stuff will be deprecated maybe in the near future. I'm just writing down what I found in this change.